### PR TITLE
Add raw web3 provider option to inject existing provider

### PR DESCRIPTION
**Pull request by howleysv**
Don't fetch gas price data when disabled

### CODE REVIEW

**Comment by levity** [comment]
@ethanbennett the GasService part of this PR looks good to me, how about you?

@howleysv thanks for the PR! is the raw provider you're wanting to inject already a subprovider that plays nice with provider-engine? if so, it might be clearer to rename things to reflect that.

**Comment by ethanbennett** [comment]
Yeah, GasService looks fine to me too. Good idea @howleysv!

**Comment by howleysv** [comment]
@levity The concrete provider object that I'm using is based on the `WebsocketProvider` from web3.js which seems to lack the `setEngine()` method to be used directly here. That's why I'm wrapping it in a `ProviderSubprovider`.

I'm totally up for renaming things for clarity, I wasn't sold on my own choices there!

**Comment by levity** [comment]
> @levity The concrete provider object that I'm using is based on the `WebsocketProvider` from web3.js which seems to lack the `setEngine()` method to be used directly here. That's why I'm wrapping it in a `ProviderSubprovider`.
> 
> I'm totally up for renaming things for clarity, I wasn't sold on my own choices there!

**Comment by howleysv** [comment]
Thanks for the feedback, I have applied the rename.

File: src/eth/accounts/setup.js [hunk 1]
```diff
@@ -28,6 +28,13 @@ export async function setupEngine(settings) {
     return new WebsocketSubprovider({ rpcUrl });
   };
 
+  const getInjectedProvider = () => {
+    if (providerSettings.inject == null) {
```
Comment (levity) [inline src/eth/accounts/setup.js]: this can be a simple `!` check instead of `== null`

**Comment by levity** [comment]
> this looks good overall, just have one comment about a style change. also, can you add a test for this? just an example of injecting a hand-crafted provider, ideally with one of its methods wrapped in `jest.fn` so that we can then verify after `await Maker.create` that the method was called.

Putting it in `test/Maker.spec.js` is fine

**Comment by howleysv** [comment]
I've added a test for initialization with the injected provider. I decided to let `Maker.authenticate()` error out and just assert it received the web3 request because it seemed like I would have to mock a bunch of web3 responses in order to have it return successfully, but if you have a better idea, lmk.
:thinking: Maybe I guess I could inject a HTTP provider and point it at ganache?

File: test/Maker.spec.js [hunk 2]
```diff
@@ -63,3 +63,19 @@ test('throws an error for an invalid id', async () => {
...
+  const maker = await Maker.create('inject', {
+    provider: { inject: mockProvider },
+    autoAuthenticate: false
+  });
+  expect(mockSend.mock.calls.length).toBe(0);
+
+  try {
+    await maker.authenticate();
+  } catch (e) {
+    expect(e.method).toBe('eth_accounts');
```
Comment (levity) [inline test/Maker.spec.js]: can you put `expect(mockSend).toBeCalled()` at this point? and `expect.assertions()` with the right number at the top of the test, so we don't have a false negative if the `authenticate` succeeds.

**Comment by howleysv** [comment]
Apologies, I haven't used jest much. Hopefully this is suitable

**Comment by levity** [comment]
> Apologies, I haven't used jest much. Hopefully this is suitable

Nothing to apologize for! Looks good.

**Pull request merged**